pydbgen/pydbgen.py

```diff
@@ -45,266 +69,151 @@ def license_plate(self,seed=None,style=None):
-                for _ in range(num):
-                    lst.append(fake.longitude())
-                return pd.Series(lst)
-            
-             
-            # Month, weekday, year, time, date
-            if data_type=='name_month':
-                for _ in range(num):
-                    lst.append(fake.month_name())
-                return pd.Series(lst)
-            if data_type=='weekday':
-                for _ in range(num):
-                    lst.append(fake.day_of_week())
-                return pd.Series(lst)
-            if data_type=='year':
-                for _ in range(num):
-                    lst.append(fake.year())
-                return pd.Series(lst)
-            if data_type=='time':
-                for _ in range(num):
-                    lst.append(fake.time())
-                return pd.Series(lst)
-            if data_type=='date':
-                for _ in range(num):
-                    lst.append(fake.date())
-                return pd.Series(lst)
-            
-            # SSN
-            if data_type=='ssn':
-                for _ in range(num):
-                    lst.append(fake.ssn())
-                return pd.Series(lst)
-            
-            # Personal, official email
-            if data_type=='email':
-                for _ in range(num):
-                    lst.append(fake.email())
-                return pd.Series(lst)
-            if data_type=='office_email':
-                for _ in range(num):
-                    lst.append(fake.company_email())
-                return pd.Series(lst)
-            
-            # Company, Job title
-            if data_type=='company':
-                for _ in range(num):
-                    lst.append(fake.company())
-                return pd.Series(lst)
-            if data_type=='job_title':
-                for _ in range(num):
-                    lst.append(fake.job())
-                return pd.Series(lst)
-            
-            # Phone number, license plate (3 styles)
-            if data_type=='phone_number_simple':
-                for _ in range(num):
-                    lst.append(self.simple_ph_num())
-                return pd.Series(lst)
-            if data_type=='phone_number_full':
-                for _ in range(num):
-                    lst.append(fake.phone_number())
-                return pd.Series(lst)
-            if data_type=='license_plate':
-                for _ in range(num):
-                    lst.append(self.license_plate())
-                return pd.Series(lst)
-
-    def gen_dataframe(self,num=10,fields=['name'], real_email=True, real_city=True, phone_simple=True, seed=None):
+            raise ValueError('Number of samples must be a positive integer, found ' + num)
+
+        if num <= 0:
+            raise ValueError('Number of samples must be a positive integer, found ' + num)
+
+        num = int(num)
+        fake = self.fake
+        fake.seed(self.seed)
+
+        func_lookup = {

```

Comment:
Again, avoids a long if-else ladder with copy-paste code, with the flexibility to add more data types easily.